src/reducers/ui.js

```diff
@@ -321,3 +331,69 @@ export const sGetUiConditionsByDimension = (state, dimension) =>
 
 export const sGetUiRepetitionByDimension = (state, dimensionId) =>
     sGetUiRepetition(state)[dimensionId]
+
+// Selector/hooks
+export const useMainDimensions = () => {
+    const store = useStore()
+    const programId = useSelector(sGetUiProgramId)
+    const inputType = useSelector(sGetUiInputType)
+
+    return useMemo(() => {
+        const { metadata } = store.getState()
+        const programType = programId && metadata[programId].programType
+
+        return Object.values(getMainDimensions()).reduce((acc, dimension) => {
+            acc[dimension.id] = {
+                ...dimension,
+                disabled: getIsMainDimensionDisabled(
+                    dimension.id,
+                    inputType,
+                    programType
+                ),
+            }
+            return acc
+        }, {})
+    }, [programId, inputType])
+}
+
+export const useMainDimension = (id) => {
+    const mainDimensions = useMainDimension()
+    return mainDimensions[id]
+}
+
+export const useTimeDimensions = () => {
+    const store = useStore()
+    const programId = useSelector(sGetUiProgramId)
+    const inputType = useSelector(sGetUiInputType)
+    const stageId = useSelector(sGetUiProgramStageId)
+
+    return useMemo(() => {
+        const { metadata } = store.getState()
+        const timeDimensions = getTimeDimensions()
+        const program = metadata[programId]
+        const stage = metadata[stageId]
+        const enabledDimensionIds = getEnabledTimeDimensionIds(
+            inputType,
+            program,
+            stage
+        )
+
+        return Object.values(timeDimensions).reduce((acc, dimension) => {
+            acc[dimension.id] = {
+                ...dimension,
+                name: getTimeDimensionName(
```

Comment:
I'm not quite sure about this part.
Is `metadata` synchronised to reflect the same name when it changes?